File: angular-ecommerce-book-store/ecommerce-book-store/src/app/common/order-item.ts
import { CartItem } from "./cart-item";

export class OrderItem {
    image: any;
    imageExtension:string | null;
    unitPrice: number;
    quantity: number;
    productId: number;

    constructor(cartItem: CartItem) {
        this.image = cartItem.image;
        this.imageExtension = cartItem.imageExtension;
        this.quantity = cartItem.quantity;
        this.unitPrice = cartItem.unitPrice;
        this.productId = cartItem.id;
    }
}

File: spring-boot-ecommerce-book-store/src/main/java/com/example/springbootecommercebookstore/entity/OrderItem.java
package com.example.springbootecommercebookstore.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.Cascade;

import java.math.BigDecimal;

@Entity
@Table(name="order_item")
@Getter
@Setter
public class OrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="id")
    private Long id;

    @Lob
    @Column(name="image")
    private byte[] image;

    @Column(name = "image_extension")
    private String imageExtension;

    @Column(name="unit_price")
    private BigDecimal unitPrice;

    @Column(name="quantity")
    private int quantity;

    @Column(name="product_id")
    private Long productId;

}
